content/client-lib-development-guide/features.textile

```diff
@@ -103,11 +103,12 @@ h3(#rest-auth). Auth
-*** @(RSA4c1)@An @ErrorInfo@ with @code@ @80019@ and description of the underlying failure should be emitted with the state change, in the @errorReason@ and/or in the callback as appropriate
-*** @(RSA4c2)@If the connection is @CONNECTING@, then the connection attempt should be treated as unsuccessful, and as such the connection should transition to the @DISCONNECTED@ or @SUSPENDED@ state as defined in "RTN14":#RTN14 and "RTN15":#RTN15
-*** @(RSA4c3)@If the connection is @CONNECTED@, then the connection should remain @CONNECTED@
-** @(RSA4d)@ If an attempt to authenticate is made using the @authUrl@ and the content type is neither @text/plain@ or @application/json@, or if using @authCallback@ and the object passed in is neither a @String@, @JsonObject@, @TokenRequest@ object or a @TokenDetails@ object, the auth attempt should fail with error code @40170@ and a suitable error message.  Additionally, if the text token string is greater than 384 bytes, or the JSON stringified @JsonObject@, @TokenRequest@ or @TokenDetails@ is greater than 128kb, then the auth attempt should fail with error code @40170@ and a suitable error message.
+** @(RSA4c)@ If an attempt by the realtime client library to authenticate is made using the @authUrl@ or @authCallback@, and the request to @authUrl@ fails, the callback @authCallback@ results in an error, the provided token is in an invalid format (as defined in "RSA4e":#RSA4e), or the attempt times out after "@realtimeRequestTimeout@":#DF1b, then:
+*** @(RSA4c1)@ An @ErrorInfo@ with @code@ @80019@ and description of the underlying failure should be emitted with the state change, in the @errorReason@ and/or in the callback as appropriate
+*** @(RSA4c2)@ If the connection is @CONNECTING@, then the connection attempt should be treated as unsuccessful, and as such the connection should transition to the @DISCONNECTED@ or @SUSPENDED@ state as defined in "RTN14":#RTN14 and "RTN15":#RTN15
+*** @(RSA4c3)@ If the connection is @CONNECTED@, then the connection should remain @CONNECTED@
+** @(RSA4d)@ If an attempt to authenticate is made using the @authUrl@ or @authCallback@ as a result of a request that uses the REST api, or by an explicit call to @authorize@ or @requestToken@, and the request to @authUrl@ fails, the callback @authCallback@ results in an error, the provided token is in an invalid format (as defined in "RSA4e":#RSA4e), or the attempt times out after "@realtimeRequestTimeout@":#DF1b, then the request should fail with error code @40170@ and a suitable error message.
+** @(RSA4e)@ The following conditions imply that the token is in an invalid format: the content type is neither @text/plain@ or @application/json@; the object passed by @authCallback@ is neither a @String@, @JsonObject@, @TokenRequest@ object or a @TokenDetails@ object; the text token string is greater than 384 bytes; or the JSON stringified @JsonObject@, @TokenRequest@ or @TokenDetails@ is greater than 128kb.
```

Comment:
`s/the content type is neither @text/plain@ or @application/json@/the @authUrl@ response content type is neither @text/plain@ or @application/json@/`